Zoom to layer bounds after double clicking on related legend item (TOC)

volterra79 opened the pull request:
In case of layer that has geometry, for example  in below image  if double click on **building**, it is possible to zoom to layer:


![Screenshot from 2023-02-01 17-05-27](https://user-images.githubusercontent.com/1051694/216280100-927eeeb8-5029-43ab-b12f-ad819366664b.png)


In case of single click,  **building** is select:

![Screenshot from 2023-02-01 17-06-06](https://user-images.githubusercontent.com/1051694/216280131-b63d3a9a-4619-420c-b248-aa82b435e564.png)

volterra79 committed:
Add click or double click for zoomTo layer in case of geolayer and has bbox property

volterra79 committed:
Add zoomToLayer on double click on TOC layer

Raruto commented on src/components/CatalogTristateTree.vue:
```diff
@@ -81,6 +81,15 @@ import GUI from 'services/gui';
+  count: 0, // count click event
+  timeoutID: null // timeoutID return by setTimeout Function
+};
+
```
1. the `@since` tag is superfluous if the variable is not exported

2. explain better the use of the `CLICKDOUBLECLICKEVENT` variable, (eg. why is it not possible to use vue attributes? `@dblclick=` and `@click=`)

volterra79 committed:
Add comments

Raruto committed:
improve comments + refactor `CLICKDOUBLECLICKEVENT`

volterra79 committed:
Fix wrong method name CLICK_EVENT.onClick (wrong) instead CLICK_EVENT.handleClick(right)
Skip if (this.isGroup || this.isTable) and not if (!this.isGroup || !this.isTable)
